modules/mobile/app/helpers/mobile/list_filter.rb

```diff
@@ -55,12 +55,19 @@ def matches
 
     def record_matches_filters?(record)
       filters.all? do |match_attribute, operations_and_values|
+        match_attribute = match_attribute.to_sym
+        model_attribute = model_attributes.find { |att| att.name == match_attribute.to_sym }
+
+        coercer = model_attribute.coercer
+
         operations_and_values.each_pair.all? do |operation, value|
+          coerced_value = coercer.call(value)
```

Comment:
The Common::Collection handles comparison by creating a mock object (which is just an instance of the model) and assigning the filter value to it, then comparing the attributes on the mock object vs the model instances in the list. This is functionally equivalent because assigning the attribute to the model uses the coercer under the hood. 

One odd behavior I noticed is that you can assign values to our models that don't match the attribute's Dry::Type. In other words, the model can specify an attribute as an Integer but you can assign a string to it. This is because `Common::Base` has this line:
`include Virtus.model(nullify_blank: true)`
when it should have this:
`include Virtus.model(nullify_blank: true, strict: true)`
IMO we should have used strict mode, but that's well outside the scope of this ticket.